src/components/AllSection.jsx

```diff
@@ -0,0 +1,143 @@
+import { useState, useEffect } from "react";
+import { Link } from "react-router-dom";
+import axios from "axios";
+import styles from "./AllSection.module.css";
+import ProductCard from "./ProductCard";
+import Pagination from "./Pagination";
+import search from "../assets/search.png";
+
+const AllSection = () => {
+  const [products, setProducts] = useState([]);
+  const [totalCount, setTotalCount] = useState(0);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [orderByOption, setOrderByOption] = useState("recent");
+  const [searchKeyword, setSearchKeyword] = useState("");
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  // 반응형 pageSize 설정
+  const [pageSize, setPageSize] = useState(10);
+
+  // 화면 크기에 따라 pageSize 변경
+  useEffect(() => {
+    const updatePageSize = () => {
+      const width = window.innerWidth;
+      if (width <= 480) {
+        setPageSize(4); // 모바일: 2열 * 2행
+      } else if (width <= 768) {
+        setPageSize(6); // 태블릿: 3열 * 2행
+      } else {
+        setPageSize(10); // 데스크톱: 5열 * 2행
+      }
+    };
+
+    updatePageSize();
+    const mobile = window.matchMedia("(max-width: 480px)");
+    const tablet = window.matchMedia("(max-width: 768px)");
+
+    const handleChange = () => updatePageSize();
+    mobile.addEventListener("change", handleChange);
+    tablet.addEventListener("change", handleChange);
+
+    return () => {
+      mobile.removeEventListener("change", handleChange);
+      tablet.removeEventListener("change", handleChange);
+    };
+  }, []);
+
+  // pageSize 변경 시 1페이지로 리셋
+  useEffect(() => {
+    setCurrentPage(1);
+  }, [pageSize]);
+
+  // API 호출: 서버에서 정렬, 검색, 페이징 처리
+  useEffect(() => {
+    const fetchProducts = async () => {
+      setLoading(true);
+      try {
+        const response = await axios.get(
+          "https://panda-market-api.vercel.app/products",
+          {
+            params: {
+              page: currentPage,
+              pageSize: pageSize,
```

Comment:
💬 여담
속성과 변수명이 동일한 경우 아래처럼 생략하셔도 됩니다~

```suggestion
              pageSize,
```